modules/common/src/main/scala/com.snowplowanalytics.snowplow.enrich/common/enrichments/registry/CurrencyConversionEnrichment.scala

```diff
@@ -216,4 +217,21 @@ final case class CurrencyConversionEnrichment[F[_]: Monad](
         )
         Monad[F].pure(FailureDetails.EnrichmentFailure(enrichmentInfo, f).invalidNel)
     }
+
+  // The original error is either a Throwable or an OER error
+  private def mkEnrichmentFailure(error: Either[Throwable, OerResponseError]): FailureDetails.EnrichmentFailure = {
+    val msg = error match {
+      case Left(t) =>
+        val sw = new StringWriter
+        t.printStackTrace(new PrintWriter(sw))
```

Comment:
Please can you explain the justification for putting the entire stack trace into the bad row?  Isn't it quite unprecedented to do this?  I mean, there must be loads of places in our estate where we catch exceptions to create bad rows, but I have never seen a stack trace get serialized before.

I'm not necessarily against it, I just wondered if this is part of a wider plan.... and how the stacktrace might get used downstream.

What are our alternatives if we don't serialize the stack trace?

Reply:
The reason is that it happened to me many times while troubleshooting the unit tests that the bad row did not contain enough information to find out what the issue was. Each time it happened I needed to update the code to print the stacktrace to identify the faulty line.

Reply:
The max allowed length of that field is https://github.com/snowplow/iglu-central/blob/master/schemas/com.snowplowanalytics.snowplow.badrows/enrichment_failures/jsonschema/2-0-0#L78

Can the strack trace exceed 512 characters?

Reply:
Good catch ! Indeed it probably can. Removing it then

Reply:
Actually I'll leave it inside and take a substring of the first 512 characters, if that's fine with you

Reply:
https://github.com/snowplow/enrich/pull/481/commits/be494ac19c4461d4b16307d8a8e44f104cb30ed8